### PR TITLE
Feature: Handling private indices

### PR DESCRIPTION
**Pull request by riendegris**
This commit adds a flag 'private' to rubber's `publish_index` function
so that a private index is not aliased by global indices, eg munin.

This flag is propagated to poi2mimir's command line, so that we can
import private POIs.

There are also modifications of the elastic search wrapper to allow
queries on non global indices.

### CODE REVIEW

File: src/bin/osm2mimir.rs [hunk 1]
```diff
@@ -126,7 +126,12 @@ fn run(args: Args) -> Result<(), mimirsbrunn::Error> {
             };
             info!("importing streets into Mimir");
             let nb_streets = rubber
-                .index(&args.dataset, &street_index_settings, streets.into_iter())
+                .index(
+                    &args.dataset,
+                    false,
```
Comment (antoine-de) [inline src/bin/osm2mimir.rs]: hum I'm not sure I'm a fan of passing this bool around as it makes the api a bit difficult to read.

Don't you think that if would be easier to add a `publish_private_index` in rubber ? (with some code factorization inside rubber)

File: tests/poi2mimir_test.rs [hunk 2]
```diff
@@ -64,48 +65,66 @@ pub fn poi2mimir_sample_test(es_wrapper: crate::ElasticSearchWrapper<'_>) {
...
+        .and_then(|s| s.keys().cloned().next())
+        .unwrap_or_else(String::new);
 
-    // Making sure that there are as many POI in ES as there are the input file.
-    assert_eq!(es_wrapper.count("_type:poi"), 2);
+    assert_eq!(poi_index_alias, "munin_poi_mti");
 
-    // We test that the POI that was close to an address has been 'attached' to that address,
-    // and inherited its admin and address: (Note that the data have been manipulated to fit)
+    //let aliases = get_first_index_aliases(raw_indexes);
```
Comment (antoine-de) [inline tests/poi2mimir_test.rs]: you can remove this line

Comment (riendegris) [reply tests/poi2mimir_test.rs]: :+1:

File: src/bin/osm2mimir.rs (hunk 1 above)
Comment (kinnou02) [reply src/bin/osm2mimir.rs]: Or something typed like an enum

Comment (riendegris) [reply src/bin/osm2mimir.rs]: agreed, using an enum

**Pull request merged**